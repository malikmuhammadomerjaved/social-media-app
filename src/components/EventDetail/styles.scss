@import "../../styles/commons.scss";

.event-detail-container {
  padding: 1rem 2rem;

  @media (min-width: $mobile-screen) {
    padding: 0;
  }

  .event-content-wrapper {
    margin-bottom: 3rem;

    @media (min-width: $mobile-screen) {
      margin-bottom: 6.5rem;
    }

    .main-heading {
      color: $primary-color;
      margin-bottom: 0.5rem;
    }

    .caption {
      color: $dark-gray-text;
    }
  }

  .events-list-wrapper {
    @include gridView(1fr, 0, 1.5rem);

    .events-list-container {
      @include flexView(center);

      .icon-tile {
        @include flexView(center, center);
        @include boxShadow(0 0 0.5rem $primary-shadow);

        width: 4.8rem;
        height: 4.8rem;
        background: $white-color;
        border-radius: 1rem;
        margin-right: 2rem;

        @media (min-width: $mobile-screen) {
          width: 5.6rem;
          height: 5.6rem;
        }

        .icon {
          width: 1.9rem;
          height: 1.9rem;

          @media (min-width: $mobile-screen) {
            width: 2.6rem;
            height: 2.6rem;
          }
        }
      }
    }

    .event-details {
      flex: 1;

      .detail-heading {
        color: $primary-color;
        margin-bottom: 0.4rem;
      }

      .detail-subtext {
        color: $dark-text;
      }
    }
  }
}
