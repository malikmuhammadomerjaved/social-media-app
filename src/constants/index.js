export const eventData = [
  {
    icon: "/images/event-page/calendar.png",
    startTime: "18 August 6:00PM",
    endTime: "19 August 1:00PM",
    utc: "UTC +10",
  },
  {
    icon: "/images/event-page/location.png",
    location: "Street name",
    address: "Suburb, State, Postcode",
  },
];
