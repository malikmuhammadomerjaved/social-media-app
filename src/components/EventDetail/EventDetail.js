import { chevronRightIcon } from "../../svg";
import "./styles.scss";

const EventDetail = (props) => {
  const { data } = props;

  return (
    <section className="event-detail-container">
      <article className="event-content-wrapper">
        <h2 className="main-heading">Birthday Bash</h2>
        <p className="caption">
          Hosted by <strong>Elysia</strong>
        </p>
      </article>

      <section className="events-list-wrapper">
        {data &&
          data.map((cardData, index) => {
            const { icon, startTime, endTime, utc, location, address } =
              cardData;
            return (
              <article key={`event-${index}`} className="events-list-container">
                <figure className="icon-tile">
                  <img src={icon} className="icon" alt="event icon" />
                </figure>

                <article className="event-details">
                  <h6 className="detail-heading">{startTime || location}</h6>
                  <p className="detail-subtext">
                    {endTime ? (
                      <span>
                        to <strong>{endTime}</strong> {utc}
                      </span>
                    ) : (
                      address
                    )}
                  </p>
                </article>

                <figure className="proceed-next">{chevronRightIcon}</figure>
              </article>
            );
          })}
      </section>
    </section>
  );
};

export default EventDetail;
