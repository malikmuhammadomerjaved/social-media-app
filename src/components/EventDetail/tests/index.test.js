import { render } from "@testing-library/react";
import EventDetail from "../EventDetail";
import { eventData } from "../../../constants";

describe("<EventDetail />", () => {
  const { container } = render(<EventDetail data={eventData} />);

  it("should match EventDetail component to snapshot", () => {
    expect(container).toMatchSnapshot();
  });
});
