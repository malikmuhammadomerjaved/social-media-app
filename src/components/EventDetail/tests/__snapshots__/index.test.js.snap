// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventDetail /> should match EventDetail component to snapshot 1`] = `
<div>
  <section
    class="event-detail-container"
  >
    <article
      class="event-content-wrapper"
    >
      <h2
        class="main-heading"
      >
        Birthday Bash
      </h2>
      <p
        class="caption"
      >
        Hosted by 
        <strong>
          Elysia
        </strong>
      </p>
    </article>
    <section
      class="events-list-wrapper"
    >
      <article
        class="events-list-container"
      >
        <figure
          class="icon-tile"
        >
          <img
            alt="event icon"
            class="icon"
            src="/images/event-page/calendar.png"
          />
        </figure>
        <article
          class="event-details"
        >
          <h6
            class="detail-heading"
          >
            18 August 6:00PM
          </h6>
          <p
            class="detail-subtext"
          >
            <span>
              to 
              <strong>
                19 August 1:00PM
              </strong>
               
              UTC +10
            </span>
          </p>
        </article>
        <figure
          class="proceed-next"
        >
          <svg
            fill="none"
            height="16"
            viewBox="0 0 9 16"
            width="9"
          >
            <path
              d="M1 1L8 8L1 15"
              stroke="#BDBDBD"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </figure>
      </article>
      <article
        class="events-list-container"
      >
        <figure
          class="icon-tile"
        >
          <img
            alt="event icon"
            class="icon"
            src="/images/event-page/location.png"
          />
        </figure>
        <article
          class="event-details"
        >
          <h6
            class="detail-heading"
          >
            Street name
          </h6>
          <p
            class="detail-subtext"
          >
            Suburb, State, Postcode
          </p>
        </article>
        <figure
          class="proceed-next"
        >
          <svg
            fill="none"
            height="16"
            viewBox="0 0 9 16"
            width="9"
          >
            <path
              d="M1 1L8 8L1 15"
              stroke="#BDBDBD"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </figure>
      </article>
    </section>
  </section>
</div>
`;
